Aliucord/src/main/java/com/aliucord/coreplugins/slashcommandsfix/Patches.java

```diff
@@ -0,0 +1,288 @@
+ * Licensed under the Open Software License version 3.0
+ */
+
+package com.aliucord.coreplugins.slashcommandsfix;
+
+import android.content.Context;
+import com.aliucord.api.GatewayAPI;
+import com.aliucord.Http;
+import com.aliucord.Logger;
+import com.aliucord.patcher.InsteadHook;
+import com.aliucord.patcher.Patcher;
+import com.aliucord.patcher.PreHook;
+import com.aliucord.Utils;
+import com.aliucord.utils.GsonUtils;
+import com.aliucord.utils.ReflectUtils;
+import com.discord.models.commands.Application;
+import com.discord.models.commands.ApplicationCommand;
+import com.discord.models.commands.ApplicationCommandKt;
+import com.discord.models.commands.ApplicationCommandLocalSendData;
+import com.discord.stores.BuiltInCommandsProvider;
+import com.discord.stores.StoreApplicationCommands;
+import com.discord.stores.StoreApplicationCommands$requestApplicationCommands$1;
+import com.discord.stores.StoreApplicationCommands$requestApplicationCommandsQuery$1;
+import com.discord.stores.StoreApplicationInteractions;
+import com.discord.stores.StoreStream;
+import com.discord.utilities.error.Error;
+import com.discord.utilities.messagesend.MessageResult;
+import com.discord.utilities.permissions.PermissionUtils;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import kotlin.jvm.functions.Function0;
+import kotlin.jvm.functions.Function1;
+
+final class Patches {
+    private ApplicationIndexCache applicationIndexCache;
+    private Logger logger;
+    private Method handleGuildApplicationsUpdateMethod;
+    private Method handleDiscoverCommandsUpdateMethod;
+    private Method handleQueryCommandsUpdateMethod;
+    private Field applicationCommandCountField;
+
+    Patches(Logger logger) throws Throwable {
+        this.logger = logger;
+        this.applicationIndexCache = new ApplicationIndexCache();
+    }
+
+    @SuppressWarnings("unchecked")
+    public void loadPatches(Context context) throws Throwable {
+        this.handleGuildApplicationsUpdateMethod = StoreApplicationCommands.class.getDeclaredMethod("handleGuildApplicationsUpdate", List.class);
+        this.handleGuildApplicationsUpdateMethod.setAccessible(true);
+        this.handleDiscoverCommandsUpdateMethod = StoreApplicationCommands.class.getDeclaredMethod("handleDiscoverCommandsUpdate", List.class);
+        this.handleDiscoverCommandsUpdateMethod.setAccessible(true);
+        this.handleQueryCommandsUpdateMethod = StoreApplicationCommands.class.getDeclaredMethod("handleQueryCommandsUpdate", List.class);
+        this.handleQueryCommandsUpdateMethod.setAccessible(true);
+        this.applicationCommandCountField = Application.class.getDeclaredField("commandCount");
+        this.applicationCommandCountField.setAccessible(true);
```

Comment:
cant you use `ReflectUtils.getMethodByArgs()` instead of this verbosity

Reply:
```
this.handleGuildApplicationsUpdateMethod = StoreApplicationCommands.class.getDeclaredMethod("handleGuildApplicationsUpdate", List.class);
this.handleGuildApplicationsUpdateMethod = ReflectUtils.getMethodByArgs(StoreApplicationCommands.class, "handleGuildApplicationsUpdate", new Object[]{List.class});
```
*This verbosity* is 26 characters less verbose than `ReflectUtils.getMethodByArgs()` lol

Reply:
1: you should not need to create new Object[], you should be able to do just `this.handleGuildApplicationsUpdateMethod = ReflectUtils.getMethodByArgs(StoreApplicationCommands.class, "handleGuildApplicationsUpdate", List.class);`

2: you wont need to setAccessible(true) for every method, since getMethodByArgs already does it

Reply:
Ok, that does make it shorter.

Reply:
For some reason, applying these changes:

```diff
diff --git a/Aliucord/src/main/java/com/aliucord/coreplugins/slashcommandsfix/Patches.java b/Aliucord/src/main/java/com/aliucord/coreplugins/slashcommandsfix/Patches.java
index 64cfd2a..b42a68c 100644
--- a/Aliucord/src/main/java/com/aliucord/coreplugins/slashcommandsfix/Patches.java
+++ b/Aliucord/src/main/java/com/aliucord/coreplugins/slashcommandsfix/Patches.java
@@ -57,12 +57,9 @@ final class Patches {
 
     @SuppressWarnings("unchecked")
     public void loadPatches(Context context) throws Throwable {
-        this.handleGuildApplicationsUpdateMethod = StoreApplicationCommands.class.getDeclaredMethod("handleGuildApplicationsUpdate", List.class);
-        this.handleGuildApplicationsUpdateMethod.setAccessible(true);
-        this.handleDiscoverCommandsUpdateMethod = StoreApplicationCommands.class.getDeclaredMethod("handleDiscoverCommandsUpdate", List.class);
-        this.handleDiscoverCommandsUpdateMethod.setAccessible(true);
-        this.handleQueryCommandsUpdateMethod = StoreApplicationCommands.class.getDeclaredMethod("handleQueryCommandsUpdate", List.class);
-        this.handleQueryCommandsUpdateMethod.setAccessible(true);
+        this.handleGuildApplicationsUpdateMethod = ReflectUtils.getMethodByArgs(StoreApplicationCommands.class, "handleGuildApplicationsUpdate", List.class);
+        this.handleDiscoverCommandsUpdateMethod = ReflectUtils.getMethodByArgs(StoreApplicationCommands.class, "handleDiscoverCommandsUpdate", List.class);
+        this.handleQueryCommandsUpdateMethod = ReflectUtils.getMethodByArgs(StoreApplicationCommands.class, "handleQueryCommandsUpdate", List.class);
         this.applicationCommandCountField = Application.class.getDeclaredField("commandCount");
         this.applicationCommandCountField.setAccessible(true);
```

Causes a `NoSuchMethodException`:

```
[PluginManager] Failed to start core plugin SlashCommandsFix
java.lang.NoSuchMethodException: com.discord.stores.StoreApplicationCommands.handleGuildApplicationsUpdate [class java.lang.Class]
  at java.lang.Class.getMethod(Class.java, [PluginManager] Failed to start core plugin SlashCommandsFix:2072)
  at java.lang.Class.getDeclaredMethod(Class.java:2050)
  at com.aliucord.utils.ReflectUtils.getMethodByArgs(ReflectUtils.java:96)
  at com.aliucord.coreplugins.slashcommandsfix.Patches.loadPatches(Patches.java:60)
  at com.aliucord.coreplugins.slashcommandsfix.SlashCommandsFix.start(SlashCommandsFix.java:23)
  at com.aliucord.coreplugins.CorePlugins.startAll(CorePlugins.kt:56)
  at com.aliucord.Main.init(Main.java:282)
  at java.lang.reflect.Method.invoke(Native Method)
  at com.aliucord.injector.InjectorKt.init(Injector.kt:116)
  at com.aliucord.injector.InjectorKt.access$init(Injector.kt:1)
  at com.aliucord.injector.InjectorKt$init$1.beforeHookedMethod(Injector.kt:43)
  at de.robv.android.xposed.XposedBridge$HookInfo.callback(XposedBridge.java:352)
```

Even though the method definitely exists, and before the changes everything seems to work correctly. Not sure what I am doing wrong.

Reply:
As mentioned by Juby on Discord, this shouldn't be used here.